src/njoy21/input/MATXSR.hpp

```diff
@@ -0,0 +1,44 @@
+class MATXSR {
+public:
+  /* nested types */
+  #include "njoy21/input/MATXSR/Card1.hpp"
+  #include "njoy21/input/MATXSR/Card2.hpp"
+  #include "njoy21/input/MATXSR/Card3.hpp"
+  #include "njoy21/input/MATXSR/Card4.hpp"
+  #include "njoy21/input/MATXSR/Card5.hpp"
+  #include "njoy21/input/MATXSR/Card6.hpp"
+  #include "njoy21/input/MATXSR/Card7.hpp"
+  #include "njoy21/input/MATXSR/Card8.hpp"
+  #include "njoy21/input/MATXSR/Card9.hpp"
+  #include "njoy21/input/MATXSR/Card10.hpp"
+
+  #include "njoy21/input/MATXSR/src/readCard4List.hpp"
+  #include "njoy21/input/MATXSR/src/readCard56List.hpp"
+  #include "njoy21/input/MATXSR/src/readCard789List.hpp"
+  #include "njoy21/input/MATXSR/src/readCard10List.hpp"
+
+  Card1 card1;
+  Card2 card2;
+  Card3 card3;
+  std::vector< Card4 > card4List;
+  std::vector< std::pair< Card5, Card6 > > card56List;
```

Comment:
Given that `Card5`--`Card9` needs to be revamped, these `std::vector`s also need to be revamped. Only `Card4` and `Card10` needs to be in a vector now.